### ISSUE TITLE
Inconsistent behaviour between save() and bulk_create() for JSON fields

### ISSUE THREAD

**Issue by nagaem** [collerek/ormar#584]
**Describe the bug**
The `Model.save()` function and the `QuerySet.bulk_create()` function differ in how they treat incoming JSON objects before saving to the database. Specifically, the `bulk_create` function always stringifies the contents of a JSON field, while the `save` function does not. This is invisible from a code perspective, as the field is re-interpreted properly on `.load()`, but it means that JSON fields from results stored in a `bulk_create` function cannot later be queried in the database.

**To Reproduce**
Steps to reproduce the behavior (with a database/metadata assigned in the BaseMeta as in the docs):
```
class TestModel(ormar.Model):
    class Meta(BaseMeta):
        tablename = "test_model"

    id: int = ormar.Integer(primary_key=True)
    json_field: Json = ormar.JSON()

# For the same json value, create two models
json_value = {"a": 1}
test_model_1 = TestModel(id=1, json_field=json_value)
test_model_2 = TestModel(id=2, json_field=json_value)

# store one with .save() and the other with .bulk_create()
await test_model_1.save()
await TestModel.objects.bulk_create([test_model_2])

# refresh from the database
await test_model_1.load()
await test_model_2.load()

assert test_model_1.json_field == test_model_2.json_field # True

# try to query the json field
table = TestModel.Meta.table
query = table.select().where(table.c.json_field["a"].as_integer() == 1)
res = [TestModel.from_row(record, source_model=TestModel) for record in await database.fetch_all(query)]

assert test_model_1 in res # True
# assert test_model_2 in res # FAILS
# assert len(res) == 2 # FAILS
```

**Expected behavior**
Both methods should behave like `.save()` - i.e. they both result in records that can be queried by JSON field.

**Screenshots**
From the pgadmin console, you can see the stringified result for test model 2.

![image](https://user-images.githubusercontent.com/17189945/156465471-c9cf7527-e10a-49cb-a5b4-3ad333f27036.png)

**Versions (please complete the following information):**
 - Database backend used: postgres (with asyncpg driver)
 - Python version: 3.10
 - `ormar` version: 0.10.25
 - `pydantic` version: 1.9.0

**Additional context**
This seems to be related to the behaviour of the `prepare_model_to_save` function, specifically the call to `dump_all_json_fields_to_str`. If this is actually a bug and not expected behaviour, I can dig around in there to try to fix it.

**Comment by RileyMShea** [comment]
I am also experiencing the described behavior.

A janky way to query stringified records in the meantime: https://stackoverflow.com/a/70878521/7262049

**Comment by collerek** [comment]
Fixed in 0.11.0 - please update and check.
